front-end/src/pages/SettingsPage.tsx

```diff
@@ -0,0 +1,84 @@
+import axios from 'axios';
+import React, { useState } from 'react';
+import {
+    Card,
+    CardBody,
+    Button,
+    Form,
+    FormGroup,
+    Input,
+    Label,
+} from 'reactstrap';
+
+const SettingsPage = (loggedInUser: any) => {
+    // https://reactstrap.github.io/components/form/#app
+    // https://medium.com/better-programming/easily-create-a-form-with-react-hooks-1cab17e2be0d
+    // setting state: https://stackoverflow.com/questions/45850550/accessing-data-from-axios-get-request
+
+    const initialInputState = { displayName: "", githubUrl: "" };
+    const authorUrl = process.env.REACT_APP_API_URL + "/api/author/" + loggedInUser.loggedInUser.authorId + "/";
+    const [eachEntry, setEachEntry] = useState(initialInputState);
+    const { displayName, githubUrl } = eachEntry;
+    const [unchangedData, setUnchangedData] = useState(initialInputState);
```

Comment:
There are ways where we can achieve the same behaviour without `unchangedData` and `setUnchangedData`.

We can leave the code as is since it works, but if you'd curious about other ways to approach this pattern, feel free to message me

Reply:
Thinking about it a bit I take it back. This isn't a pattern that I know the exact code for